### PR TITLE
Support unclosed parentheses at end of sizes attribute

### PR DESCRIPTION
**Pull request by Ahmad-S792**
<pre>
Support unclosed parentheses at end of sizes attribute
<a href="https://bugs.webkit.org/show_bug.cgi?id=254658">https://bugs.webkit.org/show_bug.cgi?id=254658</a>

Reviewed by NOBODY (OOPS!).

This patch is to align WebKit with Gecko / Firefox and Blink / Chromium.

Merge - <a href="https://chromium.googlesource.com/chromium/src.git/+/8dbf8120062e8edddc1779a3b314f2f4957610e7">https://chromium.googlesource.com/chromium/src.git/+/8dbf8120062e8edddc1779a3b314f2f4957610e7</a>

Instead of returning false, match unclosed left-parentheses and function tokens.
This progresses about eight test cases (two each) in the WPT test suite.

According to HTML Spec for Parsing Size attribute runs [1] and 
calc() parsing [2][3] a component value of it with function ending with EOF-token is valid.

[1] <a href="https://html.spec.whatwg.org/multipage/images.html#parse-a-sizes-attribute">https://html.spec.whatwg.org/multipage/images.html#parse-a-sizes-attribute</a>
[2] <a href="https://www.w3.org/TR/css-syntax-3/#parse-a-component-value">https://www.w3.org/TR/css-syntax-3/#parse-a-component-value</a>
[3] <a href="https://www.w3.org/TR/css-syntax-3/#consume-a-component-value">https://www.w3.org/TR/css-syntax-3/#consume-a-component-value</a>

Refer to following WPT issue for background: <a href="https://github.com/web-platform-tests/wpt/issues/3066">https://github.com/web-platform-tests/wpt/issues/3066</a>

* Source/WebKit/Source/core/css/parser/SizesCalcParser.cpp:
(SizesCalcParser::calcToReversePolishNotation): Update for 'unclosed' parentheses
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-1.html: Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-1-expected.txt: Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-2.html Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-2-expected.txt: Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-3.html: Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-3-expected.txt: Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-4.html: Deleted
* LayoutTests/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-4-expected.txt: Deleted
* LayoutTests/platform/ios/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-1-expected.txt: Deleted
* LayoutTests/platform/ios/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-2-expected.txt: Deleted
* LayoutTests/platform/ios/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-3-expected.txt: Deleted
* LayoutTests/platform/ios/fast/dom/HTMLImageElement/sizes/image-sizes-w3c-4-expected.txt: Deleted
* LayoutTests/TestExpectations: Remove WPT Test Exceptions
* LayoutTests/platform/mac/TestExpectations:Remove WPT Test Exceptions
* LayoutTests/platform/glib/TestExpectations: Remove WPT Test Exceptions
* LayoutTests/platform/ios-simulator/TestExpectations: Remove WPT Test Exceptions
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-display-none-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-quirks-mode-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-standards-mode-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-width-1000px-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-display-none-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-quirks-mode-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-standards-mode-expected.txt: Add Platform Specific Expectations
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-width-1000px-expected.txt: Add Platform Specific Expectations
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-display-none-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-quirks-mode-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-standards-mode-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-width-1000px-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/sizes/parse-a-sizes-attribute-expected.txt: Deleted (Without test and it is not on WPT)
</pre>

https://github.com/WebKit/WebKit/commit/751ed06ce8f3330cfabc2aef3760063d7484d6ce

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1447 "Failed to checkout and rebase branch from PR 12106") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1477 "Failed to checkout and rebase branch from PR 12106") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1527 "Failed to checkout and rebase branch from PR 12106") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2369 "Failed to checkout and rebase branch from PR 12106") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/1430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1538 "Failed to checkout and rebase branch from PR 12106") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1537 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/2369 "Failed to checkout and rebase branch from PR 12106") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/1460 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/23/builds/1538 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/14/builds/1527 "Failed to checkout and rebase branch from PR 12106") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2210 "Failed to checkout and rebase branch from PR 12106") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/23/builds/1538 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/14/builds/1527 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/2210 "Failed to checkout and rebase branch from PR 12106") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1319 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/14/builds/1527 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/2210 "Failed to checkout and rebase branch from PR 12106") | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1372 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/16/builds/1537 "Failed to checkout and rebase branch from PR 12106") | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1290 "Failed to checkout and rebase branch from PR 12106") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/14/builds/1527 "Failed to checkout and rebase branch from PR 12106") | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1414 "Failed to checkout and rebase branch from PR 12106") | | | 
| | | | |

### CODE REVIEW

**Comment by Ahmad-S792** [comment]
@rniwa - We can get rid of failing tests, because we have "WPT" tests for those and then we are right now don't have right expectations for them and have platform specific bits. I can delete 'four' of those and remove 'test expectations' and update latest expectations for 'WPT' tests. It will make it easier to track for future and make test suite cleaner as well.

**Comment by brentfulgham** [comment]
If there are WPT that cover the same things as our 'fast/dom' test cases, I would be in favor of removing our old things and importing the new ones. But if there are parts of our old tests that are not present in WPT, we should make sure we synchronize upstream WPT with those tests before losing our own checks.

File: Source/WebCore/css/parser/SizesCalcParser.cpp [hunk 1]
```diff
@@ -187,12 +187,12 @@ bool SizesCalcParser::calcToReversePolishNotation(CSSParserTokenRange range)
...
+        // If the operator token on the top of the stack is a parenthesis, then there are unclosed parentheses.
         CSSParserTokenType type = stack.last().type();
-        if (type == LeftParenthesisToken || type == FunctionToken)
-            return false;
-        // Pop the operator onto the output queue.
-        appendOperator(stack.last());
+        if (type != LeftParenthesisToken && type != FunctionToken) {
+            // Pop the operator onto the output queue.
+            appendOperator(stack.last());
+        }
```
Comment (brentfulgham) [inline Source/WebCore/css/parser/SizesCalcParser.cpp:195]: This seems fine to me, but I think Antti or Ryosuke (or Matthieu!) should double-check.

**Comment by Ahmad-S792** [comment]
> If there are WPT that cover the same things as our 'fast/dom' test cases, I would be in favor of removing our old things and importing the new ones. But if there are parts of our old tests that are not present in WPT, we should make sure we synchronize upstream WPT with those tests before losing our own checks.

Yes - I am also not super sure on whether our internal tests cover all WPT test cases. I think WPT test cases have better coverage in general. Right now, WPT tests are imported but skipped [ Failure Pass ] for three WPT tests while one is per platform level but same [ Failure Pass ]. We can remove all expectations and then update latest expectations in one go for all WPT tests and if these are not needed, can also delete `fast/dom` ones.

**Review by anttijk — APPROVED** [review]

**Comment by anttijk** [comment]
Feel free to delete any tests that are already covered by WPTs.

**Comment by Ahmad-S792** [comment]
Closing PR due to wrong rebasing on my end. Will create new.

**Pull request closed without merging**